nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -203,37 +204,52 @@ def data_generator(file, count_max):
+                    )
                 else:
-                    cap = cv2.VideoCapture(str(file))
-                    mov = []
-                    with tqdm(
-                        desc=f"Reading movie data for {Path(file).name}",
-                        position=tqdm_pos,
-                        total=total_frames,
-                        mininterval=tqdm_mininterval,
-                    ) as pbar:
-                        for _ in range(min(count_max, total_frames)):
-                            success, frame = cap.read()
-                            mov.append(frame)
-                            pbar.update(1)
-                    cap.release()
-                    image_series_kwargs.update(
-                        data=H5DataIO(
-                            DataChunkIterator(
-                                tqdm(
-                                    iterable=np.array(mov),
-                                    desc=f"Writing movie data for {Path(file).name}",
-                                    position=tqdm_pos,
-                                    mininterval=tqdm_mininterval,
-                                ),
-                                iter_axis=0,  # nwb standard is time as zero axis
-                                maxshape=tuple(maxshape),
+                    iterable = []
+                    with VideoCaptureContext(str(file)) as video_capture_ob:
+                        with tqdm(
+                            desc=f"Reading movie data for {Path(file).name}",
+                            position=tqdm_pos,
+                            total=total_frames,
+                            mininterval=tqdm_mininterval,
+                        ) as pbar:
+                            for frame in video_capture_ob:
+                                iterable.append(frame)
+                                pbar.update(1)
+                    data = H5DataIO(
+                        DataChunkIterator(
+                            tqdm(
+                                iterable=np.array(iterable),
+                                desc=f"Writing movie data for {Path(file).name}",
+                                position=tqdm_pos,
+                                mininterval=tqdm_mininterval,
                             ),
-                            compression="gzip",
-                            chunks=best_gzip_chunk,
-                        )
+                            iter_axis=0,  # nwb standard is time as zero axis
+                            maxshape=tuple(maxshape),
+                        ),
+                        compression="gzip",
+                        compression_opts=compression_options,
+                        chunks=best_gzip_chunk,
                     )
+
+                # capture data in kwargs:
+                image_series_kwargs.update(data=data)
+                if len(starting_times) != len(file_paths):
+                    starting_times.append(timestamps[-1])
```

Comment:
```suggestion
```
Looks like this is being asserted at the start of the method and so can never enter the logic.

Reply:
yes we dont need to do that
https://github.com/catalystneuro/nwb-conversion-tools/pull/372/commits/afbde8305f560b1e40a4c3aad1848e2e24b2d77a